src/vm/functions/conversions.rs

```diff
@@ -113,3 +115,107 @@ pub fn native_buff_to_uint_be(value: Value) -> Result<Value> {
     }
     return buff_to_int_generic(value, EndianDirection::BigEndian, convert_to_uint_be);
 }
+
+pub fn native_string_to_int_generic(
+    value: Value,
+    conversion_fn: fn(String) -> Result<Value>,
+) -> Result<Value> {
+    match value {
+        Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData { data }))) => {
+            let as_string = String::from_utf8(data).unwrap();
+            return conversion_fn(as_string);
+        }
+        Value::Sequence(SequenceData::String(CharType::UTF8(UTF8Data { data }))) => {
+            let flat = data.into_iter().flatten().collect();
+            let as_string = String::from_utf8(flat).unwrap();
+            return conversion_fn(as_string);
+        }
+        _ => {
+            return Err(CheckErrors::UnionTypeValueError(
+                vec![
+                    TypeSignature::max_string_ascii(),
+                    TypeSignature::max_string_utf8(),
+                ],
+                value,
+            )
+            .into())
+        }
+    };
+}
+
+pub fn native_string_to_int(value: Value) -> Result<Value> {
+    fn convert_to_int(raw_string: String) -> Result<Value> {
+        let possible_int = raw_string.parse::<i128>();
+        match possible_int {
+            Ok(val) => return Ok(Value::Int(val)),
+            Err(error) => return Err(CheckErrors::InvalidCharactersDetected.into()),
+        }
+    }
+    return native_string_to_int_generic(value, convert_to_int);
+}
+
+pub fn native_string_to_uint(value: Value) -> Result<Value> {
+    fn convert_to_uint(raw_string: String) -> Result<Value> {
+        let possible_int = raw_string.parse::<u128>();
+        match possible_int {
+            Ok(val) => return Ok(Value::UInt(val)),
+            Err(error) => return Err(CheckErrors::InvalidCharactersDetected.into()),
+        }
+    }
+    return native_string_to_int_generic(value, convert_to_uint);
+}
+
+pub fn native_int_to_string_generic(
+    value: Value,
+    conversion_fn: fn(String) -> Value,
+) -> Result<Value> {
+    match value {
+        Value::Int(ref int_value) => {
+            let as_string = int_value.to_string();
+            let value = conversion_fn(as_string);
+            return Ok(value);
+        }
+        Value::UInt(ref uint_value) => {
+            let as_string = uint_value.to_string();
+            let value = conversion_fn(as_string);
+            return Ok(value);
+        }
+        _ => {
+            return Err(CheckErrors::UnionTypeValueError(
+                vec![TypeSignature::IntType, TypeSignature::UIntType],
+                value,
+            )
+            .into())
+        }
+    };
+}
+
+pub fn native_int_to_ascii(value: Value) -> Result<Value> {
+    // Given a string representing an integer, convert this to Clarity ASCII value.
+    fn convert_to_ascii(raw_string: String) -> Value {
```

Comment:
Instead of creating a bespoke consensus-critical conversion routine from a utf8 literal (because in Rust, all Strings are utf8) to a `CharType::ASCII()`, can you just use the existing `Value::string_ascii_from_bytes()` routine, and convert the `InterpreterResult` into a `response`?

Reply:
yes.. good idea.. thanks!